pkg/network/ebpf/c/sockfd-probes.h

```diff
@@ -0,0 +1,101 @@
+#ifndef __SOCKFD_PROBES_H
+#define __SOCKFD_PROBES_H
+
+#include "ktypes.h"
+#include "bpf_builtins.h"
+#include "map-defs.h"
+
+#ifndef COMPILE_CORE
+#include <linux/ptrace.h>
+#include <linux/net.h>
+#endif
+
+#include "sock.h"
+#include "sockfd.h"
+
+SEC("kprobe/sockfd_lookup_light")
+int kprobe__sockfd_lookup_light(struct pt_regs *ctx) {
+    int sockfd = (int)PT_REGS_PARM1(ctx);
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+
+    // Check if have already a map entry for this pid_fd_t
+    // TODO: This lookup eliminates *4* map operations for existing entries
+    // but can reduce the accuracy of programs relying on socket FDs for
+    // processes with a lot of FD churn
+    pid_fd_t key = {
+        .pid = pid_tgid >> 32,
+        .fd = sockfd,
+    };
+    struct sock **sock = bpf_map_lookup_elem(&sock_by_pid_fd, &key);
```

Comment:
seems like the map `sock_by_pid_fd` is not accessed by NPM any more, if that's the case, please move it to USM files, and remove definitions and usages from NPM

Reply:
They are used from `tcp_close` here: https://github.com/DataDog/datadog-agent/blob/main/pkg/network/ebpf/c/sockfd.h#L22

Reply:
seems to be right, in that case I think we should do the following:
1. move all maps, structs, methods to files owned by USM
2. introduce a new constant (usm_tls_monitoring_enabled or a better name), and in the kernel to check we're calling the cleanup (or any other functionality in NPM) only if USM https monitoring is enabled (to spare the redundancy for the cases where usm is disabled of TLS monitoring is turned off


I'm fine with merging the current PR as is and we can take care of the followups

Reply:
(1.) is taken care of. I think we can leave (2.) for a future PR.

Reply:
perfect thanks